### PR TITLE
Initial storybooks

### PR DESCRIPTION
**Pull request by maael**
Extending @bwilson-bw's work to allow typescript and inplace (next to the component) storybooks.

* [TSDocgenPlugin](https://www.npmjs.com/package/react-docgen-typescript-webpack-plugin) Used to allow nicer storybook info usage, pulling info from typescript about component props
* [ForkTsCheckerWebpackPlugin](https://www.npmjs.com/package/fork-ts-checker-webpack-plugin) Forks typescript checking into separate processes, speeding up the check/compilation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,10 +6,12 @@
   "license": "MIT",
   "scripts": {
     "dev": "NODE_ENV=development node server",
-    "build": "next build",
```
Comment (bwilson-bw) [inline package.json]: Don't we want this as a part of the build process? Just curious

Comment (maael) [reply package.json]: It is, it's extended to include `build-storybook` too.

Comment (bwilson-bw) [reply package.json]: was`build: next build && build-storybook` always there if so then I don't know what I was talking about.

Comment (maael) [reply package.json]: It was on the additions, but the deletions it had the line as deleted. That's potentially where the confusion came from? 😛

**Pull request merged**